packages/tabs/addon/components/nucleus-tabs/tab-list-item.js

```diff
@@ -0,0 +1,267 @@
+import { once } from '@ember/runloop';
+import { TABS_KEY_CODE } from '../../constants/nucleus-tabs'
+
+/**
+  __Usage:__
+
+  [Refer component page](/docs/components/nucleus-tabs)
+
+  @class Nucleus Tab List Item
+  @namespace Components
+  @extends Ember.Component
+  @private
+*/
+@tagName('button')
+@templateLayout(layout)
+@classNames('nucleus-tabs__list__item')
+@classNameBindings('isActive:active')
+@classNameBindings('isDisabled:disabled')
+@attributeBindings('tabindex')
+@attributeBindings('role')
+@attributeBindings('aria-controls')
+@attributeBindings('aria-selected')
+@attributeBindings('isDisabled:disabled')
+@attributeBindings('title')
+class TabListItem extends Component {
+
+  /**
+  * disabled
+  *
+  * @field disabled
+  * @type string
+  * @default null
+  * @readonly
+  * @public
+  */
+  @defaultProp
+  disabled = null;
+
+  /**
+  * controls
+  *
+  * @field controls
+  * @description idref to what panel this tab item controls
+  * @type string|null
+  * @readonly
+  * @public
+  */
+  @defaultProp
+  controls;
+
+  /**
+  * role
+  *
+  * @field role
+  * @type string
+  * @default 'tab'
+  * @public
+  */
+  role = 'tab';
+
+  /**
+  * tabOrder
+  *
+  * @field tabOrder
+  * @description order in which tabs are displayed. Only first tabs will not have tabIndex value
+  * @type number
+  * @readonly
+  * @public
+  */
+  tabOrder;
+
+  /**
+  * currentSelected
+  *
+  * @field currentSelected
+  * @description currently selected tab
+  * @readonly
+  * @public
+  */
+  currentSelected;
+
+  /**
+  * tabindex
+  *
+  * @field tabindex
+  * @type string|null
+  * @public
+  */
+  @computed('tabOrder', function() {
+    return (get(this, 'tabOrder') === 0)? null : '-1';
+  })
+  tabindex;
+
+  /**
+  * isActive
+  *
+  * @field isActive
+  * @type boolean
+  * @public
+  */
+  @computed('currentSelected', function() {
+    return (get(this, 'currentSelected') === get(this, 'name'));
+  })
+  isActive;
+
+  /**
+  * isDisabled
+  *
+  * @field isDisabled
+  * @type boolean
+  * @public
+  */
+  @computed('disabled', function() {
+    return (get(this, 'disabled') === 'true')? true : false;
+  })
+  isDisabled;
+
+  /**
+  * title
+  *
+  * @field title
+  * @type string
+  * @public
+  */
+  @computed('name', function() {
+    return get(this, 'name');
+  })
+  title;
+
+  /**
+  * aria-controls
+  *
+  * @field aria-controls
+  * @type string
+  * @public
+  */
+  @computed('controls', function() {
+    return get(this, 'controls');
+  })
+  "aria-controls";
+
+  /**
+  * aria-selected
+  *
+  * @field aria-selected
+  * @type boolean
+  * @public
+  */
+  @computed('currentSelected', function() {
+    return (get(this, 'currentSelected') === get(this, 'name')).toString();
+  })
+  "aria-selected";
+
+  /**
+  * init
+  *
+  * @method init
+  * @description lifecycle event
+  * @public
+  *
+  */
+  init() {
+    super.init(...arguments);
+    once(this, get(this, 'registerTabListItem'), {
```

Comment:
Is a run loop necessary here?

Reply:
Removed using run loops. Was able to achieve the same result by using lifecycle event. Updated.